file_writer_control/JobStatus.py

```diff
@@ -0,0 +1,101 @@
+from enum import Enum, auto
+from datetime import datetime
+
+
+class JobState(Enum):
+    """
+    The state of a job.
+    """
+    NO_JOB = auto()
+    WAITING = auto()
+    WRITING = auto()
+    TIMEOUT = auto()
+    ERROR = auto()
+    DONE = auto()
+    UNAVAILABLE = auto()
+
+
+class JobStatus:
+    """
+    Contains general information about the (execution) of a job.
+    """
+    def __init__(self, job_id: str):
+        self._job_id = job_id
+        self._service_id = ""
+        self._last_update = datetime.now()
+        self._state = JobState.WAITING
+        self._message = ""
+
+    def update_status(self, new_status):
+        """
+        Updates the status/state of a this instance of the JobStatus class, using another instance.
+        .. note:: The job identifier of this instance and the other must be identical.
+        :param new_status: The other instance of the JobStatus class.
```

Comment:
Is this our convention for parameter doc-strings?  I mean I prefer:

`def foo(x):
       Args:
             x: bla bla
`
Is there a specific reason? There is nothing wrong with it of course.

Reply:
I code too little in Python to have a strong preference yet but looking at some of our other Python projects ([e.g. just-bin-it](https://github.com/ess-dmsc/just-bin-it/blob/master/just_bin_it/endpoints/config_listener.py)), this is the convention that seems to be in use.

Reply:
Yes, this is our convention. It is still the most common way of doing docstrings in the Python eco-system as it comes from Sphinx.

Also, PyCharm does it out of the box!

Reply:
Not out of the box, you have to change one setting ;)